src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -430,7 +431,7 @@ private protected override void SaveModel(ModelSaveContext ctx)
                 var buffer = Session.graph.ToGraphDef(status);
                 ctx.SaveBinaryStream("TFModel", w =>
                 {
-                    w.WriteByteArray(buffer.MemoryBlock.ToArray());
+                    w.WriteByteArray(buffer.DangerousMemoryBlock.ToArray());
```

Comment:
@Oceania2018 
Can you confirm the replacement of `MemoryBlock` here is correct?

Reply:
Correct.

Reply:
I don't think this is correct. The code needs a `using` statement around `buffer` to ensure it does not get collected before this call completes. Direct use of `DangerousMemoryBlock` will not prevent the GC from releasing it while it is still in use.

Reply:
# Resolved